platform/FileHandle.h

```diff
@@ -114,10 +114,10 @@ class FileHandle {
      *
      *  @return         Size of the file in bytes
      */
-    virtual size_t size()
+    virtual off_t size()
```

Comment:
> I was considering splitting out the off_t into seperate pr. Would that help visibility of the change to users?

Yes, definitely. The question was regarding FileHandle that it is in platform code, so if we are breaking something there ,there should be a really good reason to do so. Thus this change wont result in breaking code that already exists?